### ISSUE TITLE
Post process & conversion queue?

### ISSUE THREAD

**Issue by Hampster82** [jrudess/streamdvr#86]
Would something like a conversion queue be possible? As opposed to trying to convert at the same time if it happens that 2 or more streams end at the same time.

I don't know if other OS' are better at this, but on Windows at least the OS is desperately trying to divide the resources between the conversions that it grinds mechanical hard drives to a crawl and slows down the conversion by a significant degree.

I just ran some quick tests and if I have 2 ts files and convert them to mkv one after another in a processed queue, it takes about 1 minute per hour of stream. These example files are 15.5GB and 7.5GB, and as a whole it took about 13 minutes to convert them. The ffmpeg speed is approximately 60x through the process.

Then if I try to convert the same files at the same time, the speeds of the two conversions are about 6x and 2x through most of the process, obviously once one of them is done the second will speed up. It took about 34 minutes for the first conversion be done and then another 7 more minutes for the second one. 

So that's about 41 minutes vs 13 minutes.

**Comment by Asynchr0nous** [comment]
I have auto convert disabled and this is one of the main reasons i have it turned off. aside from possibility of things going wrong the process would be alot slower than if you were to do it manually later on. IMO not all files need converting anyway so why not just do it for files you're gonna keep (later on). I do agree this would be a great feature to have and perhaps should be made for ALL post processes and not just converting (e.g. to save bandwidth if your using the custom script to upload to cloud). this would introduce another issue which is what happens if you do ctril-c mid post-process?

**Comment by Hampster82** [comment]
What's the best way to turn the conversion off?

**Comment by jrudess** [comment]
A conversion should be pretty simple to implement

**Comment by jrudess** [comment]
Ah, I maybe spoke too soon.  A conversion-queue per site is simple to implement.  It will be slightly more work for the conversion queue to be global.  But global is really what makes the most sense.   It should also give control of how many parallel jobs to run:  0 no-limit, 1 job at a time (default), 2-n for more jobs.

I'll probably commit it to be per-site for now, and will need to think about how best to pull some of the per-site code up a level.

**Comment by jrudess** [comment]
I've committed per-site forced serialization (one job at a time).

**Comment by Hampster82** [comment]
> 
> 
> what errors? i leave mine blank. if you're getting cannot convert to .... error then its normal. its to with probem with models or sites connection and not the script

Actually I think errors are unrelated and seem to happen regardless of conversion format, just haven't noticed them before. Seems to have something to do with logging and/or drawing the tui, should probably try and reproduce and then make a separate issue ticket.

**Comment by jrudess** [comment]
Post processing is now global and all sites serialize in the post process queue.
